File: DailyChallenge/Python/20260108_SortedArray.py
def is_sorted(arr):
    ascend = "Ascending"
    descend = "Descending"
    not_sorted = "Not sorted"

    direction = None

    prevNumber = arr[0]

    for number in arr[1:]:
        #If direction hasn't been determined (first loop), determine it
        if not direction:
            direction = ascend if prevNumber <= number else descend
        #If the current number doesn't follow the determined direction, set direction to "Not sorted" and terminate loop
        elif (
            (direction == ascend and prevNumber > number) 
            or (direction == descend and prevNumber < number)
        ):
            direction = not_sorted
            break

        prevNumber = number


    return direction

File: DailyChallenge/JavaScript/20260108_SortedArray.js
function isSorted(arr) {
  let ascend = 'Ascending';
  let descend = 'Descending';
  let notSorted = 'Not sorted';

  let direction;

  let prevNumber = arr[0];

  for(let i = 1; i < arr.length; i++) {    
    let number = arr[i];

    //If direction hasn't been determined (first loop), determine it
    if(!direction){
      direction = prevNumber <= number ? ascend : descend;
    }
    //If the current number doesn't follow the determined direction, set direction to "Not sorted" and terminate loop
    else if(
		(direction == ascend && prevNumber > number) 
		|| (direction == descend && prevNumber < number)
	) {
      direction = notSorted;
      break;
    }

    prevNumber = number;
  }

  return direction;
}